tldw_Server_API/app/core/DB_Management/ChaChaNotes_DB.py

```
@@ -11029,6 +11113,202 @@ def soft_delete_persona_profile(
+            prepared_query, prepared_params = self._prepare_backend_statement(query, params)
+            cursor = conn.execute(prepared_query, prepared_params or ())
+            if cursor.rowcount > 0:
+                return True
+
+            final_state = conn.execute(
+                "SELECT version, deleted FROM persona_profiles WHERE id = ? AND user_id = ?",
+                (persona_id, user_id),
+            ).fetchone()
+            if final_state and not self._as_bool(final_state["deleted"]):
+                return True
+            return False
```

Ruff (0.15.7)

[warning] Return the condition directly

Inline condition

(SIM103)